prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiver.kt

```diff
@@ -97,94 +91,31 @@ class FHIRReceiver(
             MDCUtils.MDCProperty.REPORT_ID to queueMessage.reportId,
             MDCUtils.MDCProperty.BLOB_URL to queueMessage.blobURL,
         )
-
-    /**
-     * Retrieves the sender based on the queue message and logs any relevant errors.
-     *
-     * @param queueMessage The queue message containing details about the report.
-     * @param actionLogger The logger used to track actions and errors.
-     * @param actionHistory The action history related to receiving the report.
-     * @return The sender, or null if the sender was not found or is inactive.
-     */
-    private fun getSender(
-        queueMessage: FhirReceiveQueueMessage,
-        actionLogger: ActionLogger,
-        actionHistory: ActionHistory,
-    ): Sender? {
-        val clientId = queueMessage.headers[clientIdHeader]
-        val sender = clientId?.takeIf { it.isNotBlank() }?.let { settings.findSender(it) }
-
-        actionHistory.trackActionParams(queueMessage.headers.toString())
-
-        // Handle case where sender is not found
-        return if (sender == null) {
-            // Send an error event
-            reportEventService.sendSubmissionProcessingError(
-                ReportStreamEventName.REPORT_NOT_RECEIVABLE,
-                TaskAction.receive,
-                "Sender is not found in matching client id: ${queueMessage.headers[clientIdHeader]}.",
-                queueMessage.reportId,
-                queueMessage.blobURL
-            ) {
-                params(
-                    actionLogger.errors.associateBy { ReportStreamEventProperties.PROCESSING_ERROR }
-                    .plus(
-                        mapOf(
-                            ReportStreamEventProperties.REQUEST_PARAMETERS to queueMessage.headers.toString(),
-                        )
+        ) {
+            val clientId = queueMessage.headers[clientIdHeader]
+            val sender = clientId?.takeIf { it.isNotBlank() }?.let { settings.findSender(it) }
+            if (sender == null) {
+                val submission =
+                    Submission(
+                        queueMessage.reportId.toString(), "Rejected",
+                        queueMessage.blobURL,
+                        "Sender not found matching client_id: ${queueMessage.headers[clientIdHeader]}"
                     )
-                )
-            }
-
-            // Insert the rejection into the submission table
-            val submission =
-                Submission(
-                    queueMessage.reportId.toString(), "Rejected",
-                    queueMessage.blobURL,
-                    "Sender not found matching client_id: ${queueMessage.headers[clientIdHeader]}"
-                )
-            submissionTableService.insertSubmission(submission)
-            null
-        } else {
-            // Handle case where sender is inactive
-            if (sender.customerStatus == CustomerStatus.INACTIVE) {
```

Comment:
Dropped this since the UP via the ReportFunction silently just drops a message.  I'm of the opinion that if a sender is not suppose to sending reports we should revoke there access rather than have a flag that gets checked so late in the process.